### ISSUE TITLE
Missed OpDecorate 

### ISSUE THREAD

**Issue by iliya-diyachkov** [KhronosGroup/LLVM-SPIRV-Backend#13]
There are missed OpDecorate in 3 tests:
OpDecorate %[[PTR_ID]] MaxByteOffset 12 in transcoding/DecorationMaxByteOffset.ll
OpDecorate %[[GlobalInvocationId:[0-9]+]] BuiltIn GlobalInvocationId and others in transcoding/builtin_vars_arithmetics.ll
OpDecorate %[[#SG_MaxSize_BI:]] BuiltIn SubgroupMaxSize and others in transcoding/builtin_vars_opt.ll

**Comment by iliya-diyachkov** [comment]
transcoding/DecorationMaxByteOffset.ll is f2cf7b9c6ba158c8080621f2ca4129b139f5e210.

It looks like two other tests need changes in SPIRVIRTranslator.cpp, so waiting for Alexander's patch before continuing with this issue.

**Comment by iliya-diyachkov** [comment]
Two new tests with missed decorations:
OpDecorate %[[Id:[0-9]+]] BuiltIn GlobalLinearId in transcoding/builtin_vars.ll  
OpDecorate %[[ALIGNMENT:[0-9]+]] Alignment 16 in transcoding/DecorationAlignment.ll

**Comment by iliya-diyachkov** [comment]
The decoration issues remain in  4 tests related to BuiltIn vars decoration:
```
transcoding/builtin_vars_arithmetics.ll
builtin_vars-decorate.ll
transcoding/builtin_vars_opt.ll
transcoding/builtin_vars.ll 
```
and one more is connected to UserSemantic "annotation_on_function" decoration in: `transcoding/GlobalFunAnnotate.ll`

**Comment by iliya-diyachkov** [comment]
After  #79 only `transcoding/GlobalFunAnnotate.ll` is failing.

**Comment by iliya-diyachkov** [comment]
All tests mentioned in the issue are passed. The last one was fixed in #157.
